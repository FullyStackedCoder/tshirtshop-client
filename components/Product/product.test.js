import ProductComponent from "./index";
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeProduct = {
  product_id: "1",
  name: "Test T-Shirt",
  description: "A test T-shirt for testing",
  price: 50.0,
  discounted_price: 40.0,
  image: "product.jpg",
  image_2: "product_2.jpg",
  thumbnail: "product_thumbnail.jpg",
  display: 1
};

describe('<Product />', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ProductComponent product={fakeProduct} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders the image properly', () => {
    const wrapper = shallow(<ProductComponent product={fakeProduct} />);
    const img = wrapper.find('img');
    expect(img.props().src).toContain(fakeProduct.image);
    expect(img.props().alt).toBe(fakeProduct.name)
  });

  it('renders and displays title and price tag', () => {
    const wrapper = shallow(<ProductComponent product={fakeProduct} />);
    const PriceTag = wrapper.find('.priceTag');
    expect(PriceTag.children().text()).toBe('$40.00');
    expect(wrapper.find('.heading').text()).toBe(fakeProduct.name);
  });
});