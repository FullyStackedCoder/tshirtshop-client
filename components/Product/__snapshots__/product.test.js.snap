// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product /> renders and matches the snapshot 1`] = `
<Link
  href={
    Object {
      "pathname": "/product",
      "query": Object {
        "id": "1",
      },
    }
  }
>
  <styles__ProductStyles>
    <img
      alt="Test T-Shirt"
      src="https://tshirtshop-apollo-prod.herokuapp.com/images/product.jpg"
    />
    <styles__Title>
      Test T-Shirt
    </styles__Title>
    <styles__PriceTag>
      $40.00
    </styles__PriceTag>
    <styles__OldPriceTag>
      <s>
        $50.00
      </s>
    </styles__OldPriceTag>
    <SickButton>
      View Product
    </SickButton>
  </styles__ProductStyles>
</Link>
`;
